packages/@aws-cdk/aws-lambda/test/function.test.ts

```diff
@@ -684,6 +685,85 @@ describe('function', () => {
     })).toThrow(/deadLetterQueue defined but deadLetterQueueEnabled explicitly set to false/);
   });
 
+  test('default function with SNS DLQ when client provides Topic to be used as DLQ', () => {
+    const stack = new cdk.Stack();
+
+    const dlTopic = new sns.Topic(stack, 'DeadLetterTopic');
+
+    new lambda.Function(stack, 'MyLambda', {
+      code: new lambda.InlineCode('foo'),
+      handler: 'index.handler',
+      runtime: lambda.Runtime.NODEJS_10_X,
+      deadLetterTopic: dlTopic,
+    });
+
+    const tpl = Template.fromStack(stack);
+    tpl.hasResourceProperties('AWS::IAM::Policy', {
+      PolicyDocument: {
+        Statement: [
```

Comment:
```suggestion
        Statement: Match.arrayWith([
```

try this. should mean that we can remove the line for `Version: '2012-10-17'`, which is irrelevant.

Reply:
Should I apply that throughout the test file?

Reply:
yes please. lets strive to test only the relevant parts